frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+{
+    let howMuch = 0;
+    while (howMuch < 5) {
+        message += howMuch + " ";
+        howMuch += 1;
+    }
+    message += "times";
+}
+
+// for-of loops are analogous to for-each loops in other languages. they
+// let you loop through collections of things.
+message = "We've seen elements ";
+let array = [0, 1, 2, 3, 4, 5];
+// trying to change the variable you bind each element to will just change
+// the variable, not the array, so we usually const-declare them for clarity
+for (const element of array) {
+    message += element + " "
+}
+// message === "We've seen elements 0 1 2 3 4 5 "
+```
+
+### Functions
+Functions let you bundle chunks of code into self-contained, parameterized units. They're essential to avoiding repetition and improving readability of code. Parameters are passed by value but referencing through that value may still make changes that are visible outside of a function. This is the same way Java works.
+```js
+// functions are declared using the function keyword.
+function sayNi() {
+    // ni!
+}
+// you call a function by using its name followed by parenthesis (())
+sayNi();
+// parameters are declared alongside a function inside its parenthesis. these
+// parameters are untyped, like any other variable.
+function useNum(num) {
+    // you can use num like a variable in here
+    num - 1;
+    num += 1;
+    num = 15;
+}
+// values are returned with a return statement
+function returnOne() {
+    let varToReturn = 1;
+    varToReturn += 1;
+    return varToReturn - 1;
+}
+// the value returned by a function is "slotted in" to the place where the
+// function was called
+// e.g. this is true, it looks like 1 === 1 to the interpreter since the
+// returnOne function returned 1
+returnOne() === 1;
+// if you omit a parameter when calling a function, its value is `undefined`
+// inside the function
+function identity(x) {
+    return x;
+}
+identity() === undefined;
+
+// if the interpreter has enough information to name a function itself, you can
+// omit the name from the initial declaration and later refer to it by the
+// context-assigned name
+let namedFunction = function() {
+    // I'm named namedFunction
+}
+namedFunction();
+let functionArray = [
+    function() { /* I'm named functionArray[0] */ },
+    function() { /* I'm named functionArray[1] */ },
+];
+functionArray[0]();
+functionArray[1]();
+
+// you can use arrow syntax functions (or 'lambdas') as more concise notation
+let otherNamedFunction = () => { /* I'm named otherNamedFunction */ };
+otherNamedFunction();
+// lambdas can do anything functions can, including taking parameters and
+// returning values
+let lambdaSuccessor = (number) => { return number + 1; };
+lambdaSuccessor(1) === 2;
+// if a lambda is able to return in one expression, you can omit the braces and
+// return statement as a shorthand. this function is equivalent to the last:
+lambdaSuccessor = (number) => number + 1;
+lambdaSuccessor(1) === 2;
+```
+
+### Objects
```

Comment:
Should we talk about getters and setters? (the JavaScript syntax sugar, not necessarily the broader concept of getters and setters)

Reply:
If JS has a shorthand for them or a very different way of writing them than Java/other introductory languages, I would argue for including getters and setters in JS.